src/HttpClient.php

```diff
@@ -91,6 +93,67 @@ public function deploy(array $data)
         $this->guzzle->post('deploy', ['json' => $data]);
     }
 
+    /**
+     * Notify Bugsnag of a build.
+     *
+     * @param array $data the build information
+     *
+     * @return void
+     */
+    public function sendBuildReport(array $buildInfo)
+    {
+        $app = $this->config->getAppData();
+
+        $data = [];
+        $sourceControl = [];
+        if (isset($buildInfo['repository'])) {
+            $sourceControl['repository'] = $buildInfo['repository'];
+        }
+
+        if (isset($buildInfo['provider'])) {
+            $sourceControl['provider'] = $buildInfo['provider'];
+        } elseif (isset($buildInfo['repository'])) {
+            $url = $buildInfo['repository'];
+            if (strpos($url, 'github.com') != false) {
```

Comment:
You dont need to do this. Provider is optional

Reply:
Every field is optional though. Are we using this field for something?

Reply:
We shouldnt automatically set this in any of our sdks. This logic of guessing the provider is on the server (and therefore is easier to update).

The user needs to be able to set it for when they are using an on-premise installation of github and the repository URL isnt the github.com one.

Reply:
It gives us the extra data they may not set otherwise.

Reply:
Ah, ok, didn't see this.